### ISSUE TITLE
Forbid connection to itself in replication

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#1056]
A simple fix from bsync:

```
+bool
 replication_join(int fd, struct xrow_header *packet,
                 void (*on_join)(const struct tt_uuid *))
 {

        struct tt_uuid server_uuid = uuid_nil;
        xrow_decode_join(packet, &server_uuid);
+       if (tt_uuid_is_equal(&server_uuid, &recovery->server_uuid))
+               tnt_raise(ClientError, ER_SERVER_ID_IS_LOCAL);
```

**Comment by kostja** [comment]
* Roman Tsisyk <notifications@github.com> [15/09/22 13:42]:
> A simple fix from bsync:
> 
> ```
> +bool
>  replication_join(int fd, struct xrow_header *packet,
>                  void (*on_join)(const struct tt_uuid *))
>  {
> 
>         struct tt_uuid server_uuid = uuid_nil;
>         xrow_decode_join(packet, &server_uuid);
> +       if (tt_uuid_is_equal(&server_uuid, &recovery->server_uuid))
> +               tnt_raise(ClientError, ER_SERVER_ID_IS_LOCAL);

OK to push.


-- 
http://tarantool.org - a NoSQL database in a Lua script

**Comment by kostja** [comment]
Roman, please close if this is fixed in applier-wait-connection.

**Comment by kostja** [comment]
This is done.

**Comment by rtsisyk** [comment]
Still not fixed.

**Comment by rtsisyk** [comment]
This problem has been fixed as part of #1062. See 70e6e55322b6a11c8212e1d4cd6dd9c986aff9fe
